CMakeLists.txt

```diff
@@ -59,6 +59,11 @@ endif()
 option(BUILD_TESTING "Create tests using CMake" OFF)
 include(CTest)
 
+option(DEBUG_OUTPUT "Print debug error messages to cerr" ON)
+if(DEBUG_OUTPUT)
+    add_compile_definitions("DEBUG_OUTPUT")
```

Comment:
I think there may be two problems with this definition: 
* It is not prefixed by any `OSQP_EIGEN_` or similar prefix, so it could be prone to macro name collision, so probably it make sense to give to it a less generic name.  
* To avoid confusion for which target this compile definition is valid, I would suggest to use `target_compile_definitions` instead of `add_compile_definitions`